common/util.go

```diff
@@ -417,18 +424,72 @@ func ValidateRetryPolicy(policy *commonpb.RetryPolicy) error {
 	return nil
 }
 
+func GetDefaultRetryPolicyConfigOptions() map[string]interface{} {
+	return map[string]interface{}{
+		"InitialIntervalInSeconds": defaultInitialIntervalInSeconds,
+		"MaximumIntervalInSeconds": defaultMaximumIntervalCoefficient,
+		"BackoffCoefficient":       defaultBackoffCoefficient,
+		"MaximumAttempts":          defaultMaximumAttempts,
+	}
+}
+
+func FromConfigToDefaultRetrySettings(options map[string]interface{}) DefaultRetrySettings {
+	defaultSettings := DefaultRetrySettings{
+		InitialIntervalInSeconds:   defaultInitialIntervalInSeconds,
+		MaximumIntervalCoefficient: defaultMaximumIntervalCoefficient,
+		BackoffCoefficient:         defaultBackoffCoefficient,
+		MaximumAttempts:            defaultMaximumAttempts,
+	}
+
+	initialIntervalInSeconds, ok := options["InitialIntervalInSeconds"]
+	if ok {
+		defaultSettings.InitialIntervalInSeconds = int32(initialIntervalInSeconds.(int))
+	}
+
+	maximumIntervalCoefficient, ok := options["MaximumIntervalCoefficient"]
+	if ok {
+		defaultSettings.MaximumIntervalCoefficient = maximumIntervalCoefficient.(float64)
+	}
+
+	backoffCoefficient, ok := options["BackoffCoefficient"]
+	if ok {
+		defaultSettings.BackoffCoefficient = backoffCoefficient.(float64)
+	}
+
+	maximumAttempts, ok := options["MaximumAttempts"]
+	if ok {
+		defaultSettings.MaximumAttempts = int32(maximumAttempts.(int))
+	}
+
+	var empty commonpb.RetryPolicy
+	EnsureRetryPolicyDefaults(&empty, defaultSettings)
+	err := ValidateRetryPolicy(&empty)
+	if err != nil {
+		panic(
```

Comment:
Usually we don't panic directly within server code.  Recommended pattern is to return an error and have proper error handling for caller of this API.  I'm actually perfectly fine removing this validation all together and just have a unit test to make sure we never introduce default retry policy which is invalid.

Reply:
Once we open up dynamic config, anybody can change these defaults at any time, which was the concern, but we can definitely just fail this during API calls instead. Happy to remove this